drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,691 @@
+
+	retval = gc9x01x_configure(dev);
+	if (retval < 0) {
+		LOG_ERR("Could not configure display (%d)", retval);
+		return retval;
+	}
+
+	retval = gc9x01x_exit_sleep(dev);
+	if (retval < 0) {
+		LOG_ERR("Could not exit sleep mode (%d)", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+static int gc9x01x_set_mem_area(const struct device *dev, const uint16_t x, const uint16_t y,
+				const uint16_t w, const uint16_t h)
+{
+	int retval;
+	uint16_t spi_data[2];
+
+	spi_data[0] = sys_cpu_to_be16(x);
+	spi_data[1] = sys_cpu_to_be16(x + w - 1U);
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_COLSET, &spi_data[0], 4U);
+	if (retval < 0) {
+		return retval;
+	}
+
+	spi_data[0] = sys_cpu_to_be16(y);
+	spi_data[1] = sys_cpu_to_be16(y + h - 1U);
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_ROWSET, &spi_data[0], 4U);
+	if (retval < 0) {
+		return retval;
+	}
+
+	return 0;
+}
+
+static int gc9x01x_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			 const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct gc9x01x_config *config = dev->config;
+	struct gc9x01x_data *data = dev->data;
+	int retval;
+	const uint8_t *write_data_start = (const uint8_t *)buf;
+	struct spi_buf tx_buf;
+	struct spi_buf_set tx_bufs;
+	uint16_t write_cnt;
+	uint16_t nbr_of_writes;
+	uint16_t write_h;
+
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller than width");
+	__ASSERT((desc->pitch * data->bytes_per_pixel * desc->height) <= desc->buf_size,
+		 "Input buffer to small");
+
+	LOG_DBG("Writing %dx%d (w,h) @ %dx%d (x,y)", desc->width, desc->height, x, y);
+	retval = gc9x01x_set_mem_area(dev, x, y, desc->width, desc->height);
+	if (retval < 0) {
+		return retval;
+	}
+
+	if (desc->pitch > desc->width) {
+		write_h = 1U;
+		nbr_of_writes = desc->height;
+	} else {
+		write_h = desc->height;
+		nbr_of_writes = 1U;
+	}
+
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_MEMWR, write_data_start,
+				  desc->width * data->bytes_per_pixel * write_h);
+	if (retval < 0) {
+		return retval;
+	}
+
+	tx_bufs.buffers = &tx_buf;
+	tx_bufs.count = 1U;
+
+	write_data_start += desc->pitch * data->bytes_per_pixel;
+	for (write_cnt = 1U; write_cnt < nbr_of_writes; ++write_cnt) {
+		tx_buf.buf = (void *)write_data_start;
+		tx_buf.len = desc->width * data->bytes_per_pixel * write_h;
+
+		retval = spi_write_dt(&config->spi, &tx_bufs);
+		if (retval < 0) {
+			return retval;
+		}
+
+		write_data_start += desc->pitch * data->bytes_per_pixel;
+	}
+
+	return 0;
+}
+
+static int gc9x01x_read(const struct device *dev, const uint16_t x, const uint16_t y,
+			const struct display_buffer_descriptor *desc, void *buf)
+{
+	LOG_ERR("Reading not supported");
+	return -ENOTSUP;
+}
+
+static void *gc9x01x_get_framebuffer(const struct device *dev)
+{
+	LOG_ERR("Direct framebuffer access not supported");
+	return NULL;
+}
+
+static int gc9x01x_set_contrast(const struct device *dev, const uint8_t contrast)
+{
+	LOG_ERR("Set contrast not supported");
+	return -ENOTSUP;
+}
```

Comment:
the stubs in `include/zephyr/drivers/display.h` already check for some non implemented functions, I think you can drop all of these

Reply:
Fixed in [a1256a9](https://github.com/zephyrproject-rtos/zephyr/commit/a1256a9156de7aca7ae9d46e4335310a8bb595bb)